packages/nextjs/components/scaffold-eth/TokenAmount.tsx

```diff
@@ -0,0 +1,64 @@
+import { formatEther } from "viem";
+import { useAccountBalance } from "~~/hooks/scaffold-eth";
+import { useTargetNetwork } from "~~/hooks/scaffold-eth/useTargetNetwork";
+
+type TokenAmountProps = {
+  amount: number;
```

Comment:
also, when I pass `Number.MAX_SAFE_INTEGER` it's only 0,009 Eth. So you can't pass number greater. I think it's better to use `BigInt`

Reply:
Yup also another reason to use `BigInt` is while reading from contract you will mostly be getting as bigint and we don't want people to convert it `number` before passing it to component